ProtoGame2D/Code/Game/Framework/App.cpp

```diff
@@ -41,12 +41,12 @@ void App::Startup()
 
 	LogStartup();
 
-	m_ShaderDef = new ShaderDefinition(*ShaderDefinition::InitializeDef("Data/Materials/shader.xml"));
-	m_Shader = Renderer::GetInstance()->GetOrCreateShader(m_ShaderDef);
-
-	m_ScreenShaderDef = new ShaderDefinition(*ShaderDefinition::InitializeDef("Data/Materials/screenShader.xml"));
+	m_ShaderDef = new ShaderDefinition(*ShaderDefinition::InitializeDef("../Data/Materials/shader.xml"));
+	m_Shader = new Shader(m_ShaderDef);
```

Comment:
App should not own or create the shader. The renderer creates it and the app or the game only holds a reference to it.  Also app is supposed to be very light weight. All of these should go to game.

Reply:
Why'd you remove getorcreateshader?